rice-namenode/source/main.cc

```diff
@@ -60,6 +61,12 @@ static inline int parse_cmdline_options(
           return -1;
         }
         break;
+      case 'n':*node_policy = optarg[0];
```

Comment:
`*node_policy` should only be assigned if `optarg[0]` is available, so to preserve the default of `MAX_FREE_SPACE`. The current behavior always sets `node_policy`.

Reply:
My understanding of the way getopt worked is that it would only set c to 'n' (and thereby execute that case of the switch statement) if a correct flag of -n <param> was present, so if no -n option was set it would preserve the default value. This is my first time using getopt though, so I could be wrong.

Reply:
Good call, that seems correct